Prepare for the Bugzilla to Jira migration

flo-renaud opened the pull request:
ipatool extracts the list of BZs fixed by the commit by looking for BZ urls in the pagure ticket rhbz value. Now that RHEL is moving from bugzilla to Jira, the pagure ticket rhbz value can also contain a Jira url, like https://issues.redhat.com/browse/RHEL-1234

Extract the Jira tickets and print them in the summary before the pr-push operation is done.

flo-renaud committed:
Prepare for the Bugzilla to Jira migration

ipatool extracts the list of BZs fixed by the commit
by looking for BZ urls in the pagure ticket rhbz value.
Now that RHEL is moving from bugzilla to Jira, the
pagure ticket rhbz value can also contain a Jira url, like
https://issues.redhat.com/browse/RHEL-1234

Extract the Jira tickets and print them in the summary
before the pr-push operation is done.

Heads-up: if you already have a ~/.ipa/toolconf.yaml
configuration file for this tool, the following line needs
to be added to the file:
jira-ticket-url: https://issues.redhat.com/browse/RHEL-

You can also generate a new sample configuration using
ipatool sample-config
copy the sample content into ~/.ipa/toolconf.yaml and
customize according to your environment.

Signed-off-by: Florence Blanc-Renaud <flo@redhat.com>

flo-renaud commented:
In order to test, your existing ~/.ipa/toolconf.yaml must be edited and the following line needs to be added:
```
jira-ticket-url: https://issues.redhat.com/browse/RHEL-
```

rcritten commented:
> In order to test, your existing ~/.ipa/toolconf.yaml must be edited and the following line needs to be added:
> 
> ```
> jira-ticket-url: https://issues.redhat.com/browse/RHEL-
> ```

Will this be required post-merge as well? If so it should probably be included in the commit message for posterity.

flo-renaud commented:
> Will this be required post-merge as well? If so it should probably be included in the commit message for posterity.

I updated the commit message. The additional line is included in the new sample config generated using "ipatool sample-config" but needs to be manually added to existing ~/.ipa/toolconf.yaml.

rcritten commented:
Thanks. Yeah in retrospect my previous question wasn't worded very well. I see it'll blow up without the config entry which is fine, we want it to be very obvious that something required is missing :-)

The code reads good to me. Do you have a sample ticket to test it against?

flo-renaud commented:
@rcritten 
I created the following JIRA ticket: https://issues.redhat.com/browse/RHEL-1659
then edited https://pagure.io/freeipa/issue/9428 and added the jira url in the "rhbz" field.
ipatool in dry-run mode seems to work, the list of jira links is properly displayed:
```
$ ipatool pr-push --dry-run 6973 --reviewer rcritten
Reviewer: Rob Crittenden <rcritten@redhat.com>
Will apply 1 patches to: ipa-4-11
Fetching...
Applying to ipa-4-11: ipatests: accommodate DST in ACME cert expiry
Trying push...
git push --dry-run origin 091a89dc6f7575fd0942788a1117ad44c9c81a4a:ipa-4-11
To ssh://pagure.io/freeipa.git
   4b1c5a5a8..091a89dc6  091a89dc6f7575fd0942788a1117ad44c9c81a4a -> ipa-4-11
→ 0
Generating info...
Retrieving ticket 9428
=== Diffstat for ipa-4-11 ===
git diff --stat --color=always origin/ipa-4-11..091a89dc6f7575fd0942788a1117ad44c9c81a4a
 ipatests/test_integration/test_acme.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)
→ 0
=== Log for ipa-4-11 ===
git log --reverse --color=always origin/ipa-4-11..091a89dc6f7575fd0942788a1117ad44c9c81a4a
commit 091a89dc6f7575fd0942788a1117ad44c9c81a4a
Author: Mohammad Rizwan <myusuf@redhat.com>
Date:   Mon Aug 21 12:51:13 2023 +0530

    ipatests: accommodate DST in ACME cert expiry
    
    There is one hour time difference between expiry of ACME cert if
    the certificate is issued while daylight saving is start and
    expires after DST ends. For 2023 daylight saving time start at
    Sunday 12 March and ends at Sunday 5 November. Every certificate
    which is expiring after November 5th will have 1 hour difference in
    expiry.
    
    Fix is to use 90days+2hours to expire the cert.
    
    Fixes: https://pagure.io/freeipa/issue/9428
    
    Signed-off-by: Mohammad Rizwan <myusuf@redhat.com>
    Reviewed-By: Rob Crittenden <rcritten@redhat.com>
→ 0
=== Patches pushed ===
/home/remote/frenaud/push/patches/0000-ipatests-accommodate-DST-in-ACME-cert-expiry.patch
=== Mail summary ===
Pushed to ipa-4-11: 091a89dc6f7575fd0942788a1117ad44c9c81a4a
=== Ticket comment ===
ipa-4-11:

* 091a89dc6f7575fd0942788a1117ad44c9c81a4a ipatests: accommodate DST in ACME cert expiry


=== Bugzilla/JIRA comment ===
Fixed upstream
ipa-4-11:
https://pagure.io/freeipa/c/091a89dc6f7575fd0942788a1117ad44c9c81a4a
=== Tickets fixed ===
https://pagure.io/freeipa/issue/9428
=== Jira tickets fixed ===
https://issues.redhat.com/browse/RHEL-1659
=== Ready to push ===
Exiting, --dry-run specified
Cleaning up
```

rcritten commented:
I merged PR 6973 and got expected output:

```
=== Mail summary ===
Pushed to ipa-4-11: b13b8fbb472ec24dfe35a690147e43aea363f3e4
=== Ticket comment ===
ipa-4-11:

* b13b8fbb472ec24dfe35a690147e43aea363f3e4 ipatests: accommodate DST in ACME cert expiry


=== Bugzilla/JIRA comment ===
Fixed upstream
ipa-4-11:
https://pagure.io/freeipa/c/b13b8fbb472ec24dfe35a690147e43aea363f3e4
=== Tickets fixed ===
https://pagure.io/freeipa/issue/9428
=== Jira tickets fixed ===
https://issues.redhat.com/browse/RHEL-1659
=== Ready to push ===

```
I think this is ready to merge, ack.
